### PR TITLE
CLN: remove kwargs from Index._simple_new

### PR DESCRIPTION
**Pull request by topper-123**
Removes kwargs from ``Index._simple_new``.

The kwargs are actually ony used by ``DateTimeIndex.difference`` and ``TimeDeltaIndex.difference``.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
nice

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -146,21 +146,18 @@ def from_range(cls, data, name=None, dtype=None):
...
+    def _simple_new(cls, values, name=None, dtype=None):
         result = object.__new__(cls)
 
         # handle passed None, non-integers
         if values is None:
             # empty
             values = range(0, 0, 1)
         elif not isinstance(values, range):
-            return Index(values, dtype=dtype, name=name, **kwargs)
+            return Index(values, dtype=dtype, name=name)
```
Comment (jbrockmendel) [inline pandas/core/indexes/range.py]: not for this PR, but it'd be nice tot see this branch gone; doesn't belong in _simple_new

File: pandas/core/indexes/timedeltas.py [hunk 2]
```diff
@@ -406,6 +406,11 @@ def intersection(self, other, sort=False):
         """
         return super().intersection(other, sort=sort)
 
+    def difference(self, other, sort=None):
+        new_idx = super().difference(other, sort=sort)
+        new_idx.freq = None
+        return new_idx
```
Comment (jbrockmendel) [inline pandas/core/indexes/timedeltas.py]: could this be shared in one of the datetimelike classes?

File: pandas/core/indexes/range.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/indexes/range.py]: Yeah, I had that in my first version, but figured that is a different issue.

File: pandas/core/indexes/timedeltas.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/indexes/timedeltas.py]: I don't think so: PeriodIndex doesn't allow ``freq=None` and ``DatetimeIndexOpsMixin`` is shared with that`. Don't know if that's a bug in PeriodIndex or as intended, as I haven't worked much with datetime objects.

My motivation with this is getting kwargs out of a basic method and make child classess just add using ``super`` + the specific stuff. That is easier to reason about IMO.

File: pandas/core/indexes/datetimes.py [hunk 3]
```diff
@@ -464,6 +464,11 @@ def _convert_for_op(self, value):
             return _to_M8(value)
         raise ValueError("Passed item and index have different timezone")
 
+    def difference(self, other, sort=None):
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimes.py]: Can you add a docstring here? (I think you can use one of the decorators to append `Index.difference.__doc__`)

(and same below)

**Review by jorisvandenbossche — COMMENTED** [review]
Small comment, looks good for the rest!

File: pandas/core/indexes/datetimes.py (hunk 3 above)
Comment (topper-123) [reply pandas/core/indexes/datetimes.py]: added.

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Pull request merged**